commons/src/main/java/io/hyscale/commons/exception/HyscaleException.java

```diff
@@ -19,45 +19,45 @@
 
 public class HyscaleException extends Exception {
 
-    private HyscaleErrorCode hyscaleErrorCode;
+    private HyscaleError hyscaleError;
     private String[] args;
     private Integer code = ToolConstants.HYSCALE_ERROR_CODE;
 
-    public HyscaleException(HyscaleErrorCode hyscaleErrorCode, String... args) {
-        super(hyscaleErrorCode.getErrorMessage());
-        this.hyscaleErrorCode = hyscaleErrorCode;
+    public HyscaleException(HyscaleError hyscaleError, String... args) {
+        super(hyscaleError.getMessage());
+        this.hyscaleError = hyscaleError;
         this.args = args;
     }
 
-    public HyscaleException(HyscaleErrorCode hyscaleErrorCode) {
-        super(hyscaleErrorCode.getErrorMessage());
-        this.hyscaleErrorCode = hyscaleErrorCode;
+    public HyscaleException(HyscaleError hyscaleError) {
+        super(hyscaleError.getMessage());
+        this.hyscaleError = hyscaleError;
     }
 
-    public HyscaleException(Throwable throwable, HyscaleErrorCode hyscaleErrorCode) {
-        super(hyscaleErrorCode.getErrorMessage(), throwable);
-        this.hyscaleErrorCode = hyscaleErrorCode;
+    public HyscaleException(Throwable throwable, HyscaleError hyscaleError) {
```

Comment:
Set the code here itself in each parameterized constructor rather than dealing at getCode() method. It makes it clean and visible on the class